include/deal.II/grid/grid_generator.h

```diff
@@ -316,6 +354,30 @@ namespace GridGenerator
                              const Point<dim>   (&corners) [dim],
                              const bool           colorize = false);
 
+  /**
+   * A subdivided parallelepiped.
+   *
+   * @param tria The Triangulation to create. It needs to be empty upon calling this
+   * function.
+   *
+   * @param origin First corner of the parallelepiped.
+   *
+   * @param axes A vector of @p dim tensors describing the length and direction of the edges
+   * from @p origin.
+   *
+   * @param subdivisions Number of subdivisions in each of the dim directions. Each
+   * entry must be positive. An empty vector is equivalent to one subdivision in
+   * each direction.
+   *
+   * @param colorize Assign different boundary ids if set to true.
+   */
+  template <int dim, int spacedim>
+  void
+  subdivided_parallelepiped (Triangulation<dim, spacedim>  &tria,
+                             const Point<spacedim> &origin,
+                             const std::vector<Tensor<1,spacedim> > &axes,
```

Comment:
`ArrayView` only makes sense if anyone would want to create a larger array and have a view into it. I don't see that happening here.

I don't care all that much one way or the other. I'll leave it up to you whether you want to address this or not.